src/components/editable-header.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect } from 'react';
+import { areEqual } from 'react-window';
+import tw from 'twin.macro';
+
+interface EditableHeaderProps {
+  value: string;
+  isEditable: boolean;
+  onChange?: (value: any) => void;
+  children: any;
+}
+export const EditableHeader = React.memo(function (props: EditableHeaderProps) {
+  const {
+    value,
+    isEditable,
+    onChange,
+    children,
+  } = props;
+
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [editedValue, setEditedValue] = React.useState(value);
+
+  useEffect(() => {
+    setEditedValue(value);
+  }, [value]);
+
+  const onSubmit = () => {
+    onChange?.(editedValue);
+    setIsEditing(false);
+  }
+
+  if (!isEditable) return children
+
+  return (
+    <div
+      css={[
+        tw`h-full w-full max-w-full flex items-center cursor-cell`,
+      ]}
+      onClick={() => setIsEditing(true)}
```

Comment:
Let's make this a `button` instead of a div (or add the aria attributes that provide the correct hinting that this is interactive).